packages/nimble-components/src/table/specs/table-column-placeholder-hld.md

```diff
@@ -0,0 +1,151 @@
+
+\*This is considered invalid data from the table's perspective and should be fixed within the client application.
+
+#### Duration column
+
+| Special-cased field values          | Cell display                                                 | Group row display  |
+| ----------------------------------- | ------------------------------------------------------------ | ------------------ |
+| `undefined`                         | column placeholder, or empty if no placeholder is configured | `"No value"`       |
+| `null`                              | column placeholder, or empty if no placeholder is configured | `"No value"`       |
+| Invalid value (e.g. `Number.NaN`)\* | \<empty cell>                                                | \<empty group row> |
+
+\*This is considered invalid data from the table's perspective, and should be fixed within the client application.
+
+#### Icon mapping column
+
+The icon mapping column will not have a configuration for a placeholder.
+
+| Special-cased field values | Cell display  | Group row display  |
+| -------------------------- | ------------- | ------------------ |
+| `undefined`                | \<empty cell> | `"No value"`       |
+| `null`                     | \<empty cell> | `"No value"`       |
+| Non-mapped value\*         | \<empty cell> | \<empty group row> |
+
+\*This is considered invalid data from the table's perspective and should be fixed within the client application.
+
+#### Text mapping column
+
+The text mapping column will not have a configuration for a placeholder.
+
+| Special-cased field values | Cell display  | Group row display  |
+| -------------------------- | ------------- | ------------------ |
+| `undefined`                | \<empty cell> | `"No value"`       |
+| `null`                     | \<empty cell> | `"No value"`       |
+| Non-mapped value\*         | \<empty cell> | \<empty group row> |
+
+\*This is considered invalid data from the table's perspective and should be fixed within the client application.
+
+### Implementation plan
+
+A column's placeholder will be stored as part of that column's `columnConfig` object. There will be no changes to the `TableColumn`, `ColumnInternals`, or `ColumnInternalsOptions` classes. The placeholder from the `columnConfig` object will be used by the cell views when rendering the cell.
+
+We will create a placeholder mixin that adds the following to columns that chose to use it:
+
+-   `placeholder` string property
+-   `placeholder` attribute
+-   abstract `placeholderChanged` function that will force columns using the mixin to implement `placeholderChanged` to update their column configuration
+
+The columns that will be updated to use this mixin are:
+
+-   TableColumnText
+-   TableColumnAnchor
+-   TableColumnNumberText
+-   TableColumnDateText
+-   TableColumnDurationText
+
+### Localization
+
+All group row placeholder strings will be localized through the table's localization provider. Those strings are:
```

Comment:
@ni/systemlink-pos @vpstoynova - When we group by a column that contains _empty_, _null_, or somehow _undefined_ values, the group row will have a friendly name (**No value**, **Empty**, or **No alias** - depending on the column datatype and the undefined value).

These group row friendly names can be overridden at the table level, but I expect we'll want to keep these consistent across applications, so I'd appreciate any feedback on these default values.

Reply:
Looks good to me.

Reply:
Talked with @TJ-G yesterday, and there are no concerns about these placeholder values.